src/gwc/src/test/java/org/geoserver/gwc/GWCIntegrationTest.java

```diff
@@ -2086,4 +2087,30 @@ private void createLayerGroup(String layerGroupName, QName... layersNames) throw
         catalogBuilder.calculateLayerGroupBounds(layerGroup);
         getCatalog().add(layerGroup);
     }
+
+    /**
+     * Test verifying that GWC integration will work when layer belongs to default workspace and
+     * layer name is passed without workspace being part of layer name or OWS request itself
+     */
+    @Test
+    public void testDirectDefaultWorkspaceWMSIntegration() throws Exception {
+        final GWC gwc = GWC.get();
+        gwc.getConfig().setDirectWMSIntegrationEnabled(true);
+        final String layerName = BASIC_POLYGONS.getLocalPart();
+
+        String request =
```

Comment:
This is not testing what the ticket reports, which is, a request on a local workspace service, aka virtual services, e.g.:

```cite/wms?service=WMS&request=GetMap&version=1.1.1&format=image/png&...```

See also: https://docs.geoserver.org/latest/en/user/configuration/virtual-services.html

Reply:
I will update the test to cover this scenario